packages/zendesk-adapter/src/config.ts

```diff
@@ -3149,3 +3176,92 @@ export const isGuideThemesEnabled = (
 ): boolean => (
   fetchConfig.guide?.themesForBrands !== undefined && fetchConfig.guide?.themesForBrands.length > 0
 )
+
+export const validateOmitInactiveConfig = (
+  omitInactiveConfig: OmitInactiveConfig | undefined,
+  adapterApiConfig: configUtils.AdapterApiConfig,
+): void => {
+  if (omitInactiveConfig !== undefined) {
+    validateDefaultWithCustomizations(
+      'omitInactive',
+      omitInactiveConfig,
+      _.defaults(
+        {},
+        adapterApiConfig,
+        {
+          supportedTypes: {
+            tag: ['tags'],
+          },
+        },
+      ),
+    )
+  }
+}
+const extractOmitInactiveConfig = (
+  apiDefinitions: ZendeskApiConfig
+): { omitInactiveConfig: OmitInactiveConfig; updatedApiDefinitions: ZendeskApiConfig } | undefined => {
+  const omitInactiveDefault = _.get(apiDefinitions.typeDefaults?.transformation, 'omitInactive')
+  const customizations: Record<string, boolean> = {}
+
+  const updatedApiDefinitions: ZendeskApiConfig = _.cloneDeep(apiDefinitions)
+  if (omitInactiveDefault !== undefined) {
+    _.unset(updatedApiDefinitions.typeDefaults.transformation, 'omitInactive')
+  }
+  if (updatedApiDefinitions.types !== undefined) {
+    Object.keys(updatedApiDefinitions.types).forEach(typeKey => {
+      const typeConfig = apiDefinitions.types[typeKey]
+      if (typeConfig.transformation && _.has(typeConfig.transformation, 'omitInactive')) {
+        customizations[typeKey] = _.get(typeConfig.transformation, 'omitInactive')
+        _.unset(updatedApiDefinitions.types[typeKey].transformation, 'omitInactive')
+      }
+    })
+  }
+
+  if (omitInactiveDefault !== undefined || Object.keys(customizations).length > 0) {
+    return {
+      omitInactiveConfig: {
+        default: omitInactiveDefault,
+        customizations,
+      },
+      updatedApiDefinitions,
+    }
+  }
+  return undefined
+}
+
+
+export const migrateOmitInactiveConfig = (
+  currentConfig: InstanceElement,
+  updatedConfig: { config: InstanceElement[]; message: string } | undefined
+): { config: InstanceElement[]; message: string } | undefined => {
+  const configToOverride = updatedConfig === undefined
+    ? currentConfig.value
+    : updatedConfig.config[0].value
+  const updatedFetchConfig = configToOverride[FETCH_CONFIG]
+  const currentApiDefinitions = configToOverride[API_DEFINITIONS_CONFIG]
+  if (updatedFetchConfig.omitInactive !== undefined || currentApiDefinitions === undefined) {
```

Comment:
in case it is undefined
```suggestion
  if (updatedFetchConfig?.omitInactive !== undefined || currentApiDefinitions === undefined) {
```